### PR TITLE
feat(cli): remove default `cjs` format

### LINKED ISSUE #1854: CLI should have no output module type defined by default
Currently when running `stc` the only module type that defaults to true is `cjs`. While this can fit some use-cases, it definitely doesn't fit all of them.

In the next major version, we can break this behavior and force the user to specify all module types they require.

Something like:
```sh
stc --modules="cjs, stcss, dts, dtsSourceMap"
```

### PR DESCRIPTION
**Pull request by tzachbon**
## BREAKING

closes #1854

### CODE REVIEW

File: packages/cli/src/build-stylable.ts [hunk 1]
```diff
@@ -81,6 +82,10 @@ export async function buildStylable(
 
             log('[Project]', projectRoot, buildOptions);
 
+            if (!hasStylableCSSOutput(buildOptions)) {
+                log(`No CSS output found for "${identifier}"`, levels.info);
```
Comment (tomrav) [inline packages/cli/src/build-stylable.ts:86]: Perhaps consider changing to something like:
`No target output declared, please provide one or more of the following target options: "cjs", "esm", "css", "stcss" or "indexFile"`

Comment (tomrav) [reply packages/cli/src/build-stylable.ts:86]: and possibly a higher severity?

**Comment by barak007** [comment]
There is another change we can make for the cli interface and that's to be able to emit output and also build index I the same command. Right now if you add indexFile it will not emit assets.

**Comment by tzachbon** [comment]
> There is another change we can make for the cli interface and that's to be able to emit output and also build index I the same command. Right now if you add indexFile it will not emit assets. Maybe should be added to the v5 issue

Agreed, I'll add it to the issue and we will change it in a different PR.

File: packages/cli/src/build-stylable.ts [hunk 2]
```diff
@@ -81,6 +82,13 @@ export async function buildStylable(
 
             log('[Project]', projectRoot, buildOptions);
 
+            if (!hasStylableCSSOutput(buildOptions)) {
+                log(
+                    `No target output declared "${identifier}", please provide one or more of the following target options: "cjs", "esm", "css", "stcss" or "indexFile"`,
```
Comment (tomrav) [inline packages/cli/src/build-stylable.ts:87]:
```suggestion
                    `No target output declared for "${identifier}", please provide one or more of the following target options: "cjs", "esm", "css", "stcss" or "indexFile"`,
```

**Review by tomrav — APPROVED** [review]

**Pull request merged**